src/main/java/heronarts/lx/midi/surface/MidiFighter64.java

```diff
@@ -0,0 +1,203 @@
+  public static final int LED_AZURE = 37;
+  public static final int LED_AZURE_HALF = 38;
+  public static final int LED_AZURE_DIM = 39;
+  public static final int LED_PERIWINKLE = 40;
+  // 41-47 are more shades of blue than I can name
+  public static final int LED_LAVENDER = 48;
+  public static final int LED_PURPLE = 49;
+  public static final int LED_PURPLE_HALF = 50;
+  public static final int LED_PURPLE_DIM = 51;
+  public static final int LED_PALE_MAGENTA = 52;
+  public static final int LED_MAGENTA = 53;
+  public static final int LED_MAGENTA_HALF = 54;
+  public static final int LED_MAGENTA_DIM = 55;
+  public static final int LED_MAGENTA_PINK = 56;
+  public static final int LED_HOT_PINK = 57;
+  public static final int LED_HOT_PINK_HALF = 58;
+  public static final int LED_HOT_PINK_DIM = 59;
+  public static final int LED_GOLDENROD = 62;
+  public static final int LED_LAWN_GREEN = 63;
+  // There are 64 more colors, but they seem like dupes.
+  // There might be a small number of slightly unique ones in that range.
+
+  // Holds the colors that will be sent out to the device. This array
+  // starts with (0,0), (0,1), ... (7,7) of the LEFT page, and then
+  // the same ordering for the RIGHT page.
+  private int[] colors;
+
+  // Converts a MIDI note from the MF64 into information about which
+  // button was pressed
+  public class Mapping {
+    // If an unexpected note is received, this is set to false
+    // and all other attributes should be ignored.
+    public boolean valid;
+    // The MF64 has left and right virtual button pages.
+    public enum Page {
+      LEFT, RIGHT;
+    }
+    public Page page;
+    // 0 is the bottom row, 7 the top.
+    public int row;
+    // 0 is the left column, 7 the right
+    public int col;
+
+    public Mapping(MidiNoteOn note) {
+      int pitch = note.getPitch();
+      int channel = note.getChannel();
+      this.valid = true;
+      if (channel == 2) {
+        this.page = Page.LEFT;
+      } else if (channel == 1) {
+        this.page = Page.RIGHT;
+      } else {
+        LX.warning("Got wild-channel MIDI note " + note);
+        this.valid = false;
+        return;
+      }
+
+      if (pitch >= 36 && pitch <= 67) {
+        this.row = (pitch / 4) - 9;
+        this.col = pitch % 4;
+      } else if (pitch >= 68 && pitch <= 99) {
+        this.row = (pitch / 4) - 17;
+        this.col = pitch % 4 + 4;
+      } else {
+        LX.warning("Got wild-pitch MIDI note " + note);
+        this.valid = false;
+      }
+    }
+  }
+
+  public void setColors(int[] newColors) {
+    if (newColors.length != 128) {
+      throw new IllegalArgumentException("newColors is of length " + newColors.length);
+    }
+    for (int i = 0; i < 128; i++) {
+      if (newColors[i] < 0 || newColors[i] > 127) {
+        throw new IllegalArgumentException("newColors contained value " + newColors[i]);
+      }
+    }
+    System.arraycopy(newColors, 0, this.colors, 0, 128);
+    this.sendColors();
```

Comment:
General LX coding style, I don't include the `this.` on method calls. For member variables I find it helpful to quickly disambiguate between arguments. But static function calls are (or should be) so rare in Java that I don't find it adds much on method invocations.